twill-yarn/src/test/java/org/apache/twill/yarn/ContainerSizeTestRun.java

```diff
@@ -67,20 +69,37 @@ public void testContainerSize() throws InterruptedException, TimeoutException, E
   @Test
   public void testMaxHeapSize() throws InterruptedException, TimeoutException, ExecutionException {
     TwillRunner runner = getTwillRunner();
+    String runnableName = "sleep";
+
     TwillController controller = runner.prepare(new MaxHeapApp())
-      // Alter the AM container size
-      .withConfiguration(Collections.singletonMap(Configs.Keys.YARN_AM_MEMORY_MB, "256"))
+      // Alter the AM container size and heap ratio
+      .withConfiguration(ImmutableMap.of(Configs.Keys.YARN_AM_MEMORY_MB, "256",
+                                         Configs.Keys.HEAP_RESERVED_MIN_RATIO, "0.65"))
+      // Use a different heap ratio and reserved memory size for the runnable
+      .withConfiguration(runnableName,
+                         ImmutableMap.of(Configs.Keys.HEAP_RESERVED_MIN_RATIO, "0.8",
```

Comment:
can you add a test that, if this config for the runnable does not have HEAP_RESERVED_MIN_RATIO, it defaults to the one above (0.65) and not the system default.